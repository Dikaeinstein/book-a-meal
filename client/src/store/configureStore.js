import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
