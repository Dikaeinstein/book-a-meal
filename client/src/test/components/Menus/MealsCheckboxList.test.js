import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedMealsCheckboxList, { MealsCheckboxList }
  from '../../../components/Menus/MealsCheckboxList';

describe('<MealsCheckboxList />', () => {
  const meals = [
    { id: 1, name: 'test meal', price: 4000 },
    { id: 2, name: 'test meal2', price: 4000 },
  ];
  const props = {
    meals,
    isFetching: false,
    handleCheck: jest.fn(),
    checkedMeals: {},
  };
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore({ meals: { isFetching: false } });
    const wrapper = shallow(<ConnectedMealsCheckboxList
      store={store}
      handleCheck={jest.fn()}
      meals={meals}
      checkedMeals={{}}
    />);
    expect(wrapper.props().isFetching).toBeFalsy();
  });
  it('should call handleCheck when meal is checked', () => {
    const handleCheck = jest.fn();
    const wrapper = shallow(<MealsCheckboxList {...{ ...props, handleCheck }} />);
    const checkBox = wrapper.find('input').first();
    checkBox.simulate('change');
    expect(handleCheck).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    let tree = shallow(<MealsCheckboxList {...props} />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<MealsCheckboxList {...{ ...props, isFetching: true }} />);
    expect(tree).toMatchSnapshot();
  });
});
