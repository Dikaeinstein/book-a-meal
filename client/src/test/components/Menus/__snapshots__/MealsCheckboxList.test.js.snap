// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealsCheckboxList /> should match snapshot 1`] = `
<React.Fragment>
  <div
    key="1"
    style={
      Object {
        "margin": ".5rem 0",
      }
    }
  >
    <label>
      <input
        name="meal-1"
        onChange={[Function]}
        onClick={[Function]}
        type="checkbox"
      />
      <span>
        test meal
      </span>
    </label>
  </div>
  <div
    key="2"
    style={
      Object {
        "margin": ".5rem 0",
      }
    }
  >
    <label>
      <input
        name="meal-2"
        onChange={[Function]}
        onClick={[Function]}
        type="checkbox"
      />
      <span>
        test meal2
      </span>
    </label>
  </div>
</React.Fragment>
`;

exports[`<MealsCheckboxList /> should match snapshot 2`] = `
<Loading
  text="fetching meals..."
>
  <Preloader
    active={true}
    color="blue"
    flashing={true}
    size="small"
  />
</Loading>
`;
